### ISSUE TITLE
[Bug] training over PASCAL does not work due to different metainfo

### ISSUE THREAD

**Issue by franchesoni** [open-mmlab/mmdetection#9195]
### Prerequisite

- [X] I have searched [Issues](https://github.com/open-mmlab/mmdetection/issues) and [Discussions](https://github.com/open-mmlab/mmdetection/discussions) but cannot get the expected help.
- [X] I have read the [FAQ documentation](https://mmdetection.readthedocs.io/en/latest/faq.html) but cannot get the expected help.
- [X] The bug has not been fixed in the [latest version (master)](https://github.com/open-mmlab/mmdetection) or [latest version (3.x)](https://github.com/open-mmlab/mmdetection/tree/dev-3.x).

### Task

I'm using the official example scripts/configs for the officially supported tasks/models/datasets.

### Branch

3.x branch https://github.com/open-mmlab/mmdetection/tree/3.x

### Environment

`collect_env.py` does not work, see #9194 , but find here environment details

Installation procedure:

```
python -m venv env_iadet
source env_iadet/bin/activate
pip install --upgrade pip
pip3 install torch torchvision --extra-index-url https://download.pytorch.org/whl/cpu
pip install -U openmim
mim install mmengine
mim install "mmcv>=2.0.0rc1"
git clone https://github.com/open-mmlab/mmdetection.git -b 3.x
cd mmdetection
pip install -v -e .
```

From `pip freeze`:
```
addict==2.4.0
certifi==2022.9.24
charset-normalizer==2.1.1
click==8.1.3
colorama==0.4.6
commonmark==0.9.1
contourpy==1.0.6
cycler==0.11.0
fonttools==4.38.0
idna==3.4
kiwisolver==1.4.4
Markdown==3.4.1
matplotlib==3.6.1
mmcv==2.0.0rc2
-e git+https://github.com/open-mmlab/mmdetection.git@3221ecf33885bcf185e1ea63ac72a861403d81b0#egg=mmdet
mmengine==0.2.0
model-index==0.1.11
numpy==1.23.4
opencv-python==4.6.0.66
openmim==0.3.2
ordered-set==4.1.0
packaging==21.3
pandas==1.5.1
Pillow==9.3.0
pycocotools==2.0.5
Pygments==2.13.0
pyparsing==3.0.9
python-dateutil==2.8.2
pytz==2022.5
PyYAML==6.0
requests==2.28.1
rich==12.6.0
six==1.16.0
tabulate==0.9.0
termcolor==2.1.0
terminaltables==3.1.10
torch==1.13.0+cpu
torchvision==0.14.0+cpu
typing_extensions==4.4.0
urllib3==1.26.12
yapf==0.32.0
```

### Reproduces the problem - code sample

Download PASCAL VOC using `torchvision`:
```python
import os
import torchvision

mmdetpath = "."
torchvision.datasets.VOCDetection(os.path.join(mmdetpath, "data"), year="2012", image_set="trainval", download=True)
torchvision.datasets.VOCDetection(os.path.join(mmdetpath, "data"), year="2007", image_set="trainval", download=True)
```

### Reproduces the problem - command or script

After downloading the dataset, run:
```bash
python tools/train.py configs/pascal_voc/ssd300_voc0712.py
```

### Reproduces the problem - error message

```bash
Traceback (most recent call last):
  File "/home/franchesoni/mine/creations/phd/projects/current/iadet/mmdetection/tools/train.py", line 120, in <module>
    main()
  File "/home/franchesoni/mine/creations/phd/projects/current/iadet/mmdetection/tools/train.py", line 116, in main
    runner.train()
  File "/home/franchesoni/mine/creations/phd/projects/current/iadet/env_iadet/lib/python3.10/site-packages/mmengine/runner/runner.py", line 1629, in train
    self._train_loop = self.build_train_loop(
  File "/home/franchesoni/mine/creations/phd/projects/current/iadet/env_iadet/lib/python3.10/site-packages/mmengine/runner/runner.py", line 1444, in build_train_loop
    loop = LOOPS.build(
  File "/home/franchesoni/mine/creations/phd/projects/current/iadet/env_iadet/lib/python3.10/site-packages/mmengine/registry/registry.py", line 421, in build
    return self.build_func(cfg, *args, **kwargs, registry=self)
  File "/home/franchesoni/mine/creations/phd/projects/current/iadet/env_iadet/lib/python3.10/site-packages/mmengine/registry/build_functions.py", line 135, in build_from_cfg
    raise type(e)(
ValueError: class `EpochBasedTrainLoop` in mmengine/runner/loops.py: class `RepeatDataset` in mmengine/dataset/dataset_wrapper.py: class `ConcatDataset` in mmengine/dataset/dataset_wrapper.py: The meta information of the 2-th dataset does not match meta information of the first dataset
```

<details>
<summary>Full output</summary>
```bash
10/31 14:17:01 - mmengine - INFO - 
------------------------------------------------------------
System environment:
    sys.platform: linux
    Python: 3.10.4 (main, Mar 31 2022, 08:41:55) [GCC 7.5.0]
    CUDA available: False
    numpy_random_seed: 518614498
    GCC: gcc (Ubuntu 11.2.0-19ubuntu1) 11.2.0
    PyTorch: 1.13.0+cpu
    PyTorch compiling details: PyTorch built with:
  - GCC 9.3
  - C++ Version: 201402
  - Intel(R) Math Kernel Library Version 2020.0.0 Product Build 20191122 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v2.6.0 (Git Hash 52b5f107dd9cf10910aaa19cb47f3abf9b349815)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - LAPACK is enabled (usually provided by MKL)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - Build settings: BLAS_INFO=mkl, BUILD_TYPE=Release, CXX_COMPILER=/opt/rh/devtoolset-9/root/usr/bin/c++, CXX_FLAGS= -fabi-version=11 -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_KINETO -DLIBKINETO_NOCUPTI -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -DEDGE_PROFILER_USE_KINETO -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Werror=non-virtual-dtor -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wunused-local-typedefs -Wno-unused-parameter -Wno-unused-function -Wno-unused-result -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Werror=cast-function-type -Wno-stringop-overflow, LAPACK_INFO=mkl, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, TORCH_VERSION=1.13.0, USE_CUDA=0, USE_CUDNN=OFF, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=OFF, USE_NNPACK=ON, USE_OPENMP=ON, USE_ROCM=OFF, 

    TorchVision: 0.14.0+cpu
    OpenCV: 4.6.0
    MMEngine: 0.2.0

Runtime environment:
    cudnn_benchmark: False
    mp_cfg: {'mp_start_method': 'fork', 'opencv_num_threads': 0}
    dist_cfg: {'backend': 'nccl'}
    seed: None
    Distributed launcher: none
    Distributed training: False
    GPU number: 1
------------------------------------------------------------

10/31 14:17:02 - mmengine - INFO - Config:
input_size = 300
model = dict(
    type='SingleStageDetector',
    data_preprocessor=dict(
        type='DetDataPreprocessor',
        mean=[123.675, 116.28, 103.53],
        std=[1, 1, 1],
        bgr_to_rgb=True,
        pad_size_divisor=1),
    backbone=dict(
        type='SSDVGG',
        depth=16,
        with_last_pool=False,
        ceil_mode=True,
        out_indices=(3, 4),
        out_feature_indices=(22, 34),
        init_cfg=dict(
            type='Pretrained', checkpoint='open-mmlab://vgg16_caffe')),
    neck=dict(
        type='SSDNeck',
        in_channels=(512, 1024),
        out_channels=(512, 1024, 512, 256, 256, 256),
        level_strides=(2, 2, 1, 1),
        level_paddings=(1, 1, 0, 0),
        l2_norm_scale=20),
    bbox_head=dict(
        type='SSDHead',
        in_channels=(512, 1024, 512, 256, 256, 256),
        num_classes=20,
        anchor_generator=dict(
            type='SSDAnchorGenerator',
            scale_major=False,
            input_size=300,
            basesize_ratio_range=(0.2, 0.9),
            strides=[8, 16, 32, 64, 100, 300],
            ratios=[[2], [2, 3], [2, 3], [2, 3], [2], [2]]),
        bbox_coder=dict(
            type='DeltaXYWHBBoxCoder',
            target_means=[0.0, 0.0, 0.0, 0.0],
            target_stds=[0.1, 0.1, 0.2, 0.2])),
    train_cfg=dict(
        assigner=dict(
            type='MaxIoUAssigner',
            pos_iou_thr=0.5,
            neg_iou_thr=0.5,
            min_pos_iou=0.0,
            ignore_iof_thr=-1,
            gt_max_assign_all=False),
        sampler=dict(type='PseudoSampler'),
        smoothl1_beta=1.0,
        allowed_border=-1,
        pos_weight=-1,
        neg_pos_ratio=3,
        debug=False),
    test_cfg=dict(
        nms_pre=1000,
        nms=dict(type='nms', iou_threshold=0.45),
        min_bbox_size=0,
        score_thr=0.02,
        max_per_img=200))
cudnn_benchmark = True
dataset_type = 'VOCDataset'
data_root = 'data/VOCdevkit/'
file_client_args = dict(backend='disk')
train_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(type='LoadAnnotations', with_bbox=True),
    dict(
        type='Expand',
        mean=[123.675, 116.28, 103.53],
        to_rgb=True,
        ratio_range=(1, 4)),
    dict(
        type='MinIoURandomCrop',
        min_ious=(0.1, 0.3, 0.5, 0.7, 0.9),
        min_crop_size=0.3),
    dict(type='Resize', scale=(300, 300), keep_ratio=False),
    dict(type='RandomFlip', prob=0.5),
    dict(
        type='PhotoMetricDistortion',
        brightness_delta=32,
        contrast_range=(0.5, 1.5),
        saturation_range=(0.5, 1.5),
        hue_delta=18),
    dict(type='PackDetInputs')
]
test_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(type='Resize', scale=(300, 300), keep_ratio=False),
    dict(type='LoadAnnotations', with_bbox=True),
    dict(
        type='PackDetInputs',
        meta_keys=('img_id', 'img_path', 'ori_shape', 'img_shape',
                   'scale_factor'))
]
train_dataloader = dict(
    batch_size=8,
    num_workers=3,
    persistent_workers=True,
    sampler=dict(type='DefaultSampler', shuffle=True),
    batch_sampler=dict(type='AspectRatioBatchSampler'),
    dataset=dict(
        type='RepeatDataset',
        times=10,
        dataset=dict(
            type='ConcatDataset',
            datasets=[
                dict(
                    type='VOCDataset',
                    data_root='data/VOCdevkit/',
                    ann_file='VOC2007/ImageSets/Main/trainval.txt',
                    data_prefix=dict(sub_data_root='VOC2007/'),
                    filter_cfg=dict(filter_empty_gt=True, min_size=32),
                    pipeline=[
                        dict(type='LoadImageFromFile'),
                        dict(type='LoadAnnotations', with_bbox=True),
                        dict(
                            type='Expand',
                            mean=[123.675, 116.28, 103.53],
                            to_rgb=True,
                            ratio_range=(1, 4)),
                        dict(
                            type='MinIoURandomCrop',
                            min_ious=(0.1, 0.3, 0.5, 0.7, 0.9),
                            min_crop_size=0.3),
                        dict(
                            type='Resize', scale=(300, 300), keep_ratio=False),
                        dict(type='RandomFlip', prob=0.5),
                        dict(
                            type='PhotoMetricDistortion',
                            brightness_delta=32,
                            contrast_range=(0.5, 1.5),
                            saturation_range=(0.5, 1.5),
                            hue_delta=18),
                        dict(type='PackDetInputs')
                    ]),
                dict(
                    type='VOCDataset',
                    data_root='data/VOCdevkit/',
                    ann_file='VOC2012/ImageSets/Main/trainval.txt',
                    data_prefix=dict(sub_data_root='VOC2012/'),
                    filter_cfg=dict(filter_empty_gt=True, min_size=32),
                    pipeline=[
                        dict(type='LoadImageFromFile'),
                        dict(type='LoadAnnotations', with_bbox=True),
                        dict(
                            type='Expand',
                            mean=[123.675, 116.28, 103.53],
                            to_rgb=True,
                            ratio_range=(1, 4)),
                        dict(
                            type='MinIoURandomCrop',
                            min_ious=(0.1, 0.3, 0.5, 0.7, 0.9),
                            min_crop_size=0.3),
                        dict(
                            type='Resize', scale=(300, 300), keep_ratio=False),
                        dict(type='RandomFlip', prob=0.5),
                        dict(
                            type='PhotoMetricDistortion',
                            brightness_delta=32,
                            contrast_range=(0.5, 1.5),
                            saturation_range=(0.5, 1.5),
                            hue_delta=18),
                        dict(type='PackDetInputs')
                    ])
            ])))
val_dataloader = dict(
    batch_size=1,
    num_workers=2,
    persistent_workers=True,
    drop_last=False,
    sampler=dict(type='DefaultSampler', shuffle=False),
    dataset=dict(
        type='VOCDataset',
        data_root='data/VOCdevkit/',
        ann_file='VOC2007/ImageSets/Main/test.txt',
        data_prefix=dict(sub_data_root='VOC2007/'),
        test_mode=True,
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(type='Resize', scale=(300, 300), keep_ratio=False),
            dict(type='LoadAnnotations', with_bbox=True),
            dict(
                type='PackDetInputs',
                meta_keys=('img_id', 'img_path', 'ori_shape', 'img_shape',
                           'scale_factor'))
        ]))
test_dataloader = dict(
    batch_size=1,
    num_workers=2,
    persistent_workers=True,
    drop_last=False,
    sampler=dict(type='DefaultSampler', shuffle=False),
    dataset=dict(
        type='VOCDataset',
        data_root='data/VOCdevkit/',
        ann_file='VOC2007/ImageSets/Main/test.txt',
        data_prefix=dict(sub_data_root='VOC2007/'),
        test_mode=True,
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(type='Resize', scale=(300, 300), keep_ratio=False),
            dict(type='LoadAnnotations', with_bbox=True),
            dict(
                type='PackDetInputs',
                meta_keys=('img_id', 'img_path', 'ori_shape', 'img_shape',
                           'scale_factor'))
        ]))
val_evaluator = dict(type='VOCMetric', metric='mAP', eval_mode='11points')
test_evaluator = dict(type='VOCMetric', metric='mAP', eval_mode='11points')
train_cfg = dict(type='EpochBasedTrainLoop', max_epochs=24, val_interval=1)
val_cfg = dict(type='ValLoop')
test_cfg = dict(type='TestLoop')
param_scheduler = [
    dict(
        type='LinearLR', start_factor=0.001, by_epoch=False, begin=0, end=500),
    dict(
        type='MultiStepLR',
        begin=0,
        end=24,
        by_epoch=True,
        milestones=[16, 20],
        gamma=0.1)
]
optim_wrapper = dict(
    type='OptimWrapper',
    optimizer=dict(type='SGD', lr=0.001, momentum=0.9, weight_decay=0.0005))
auto_scale_lr = dict(enable=False, base_batch_size=64)
default_scope = 'mmdet'
default_hooks = dict(
    timer=dict(type='IterTimerHook'),
    logger=dict(type='LoggerHook', interval=50),
    param_scheduler=dict(type='ParamSchedulerHook'),
    checkpoint=dict(type='CheckpointHook', interval=1),
    sampler_seed=dict(type='DistSamplerSeedHook'),
    visualization=dict(type='DetVisualizationHook'))
env_cfg = dict(
    cudnn_benchmark=False,
    mp_cfg=dict(mp_start_method='fork', opencv_num_threads=0),
    dist_cfg=dict(backend='nccl'))
vis_backends = [dict(type='LocalVisBackend')]
visualizer = dict(
    type='DetLocalVisualizer',
    vis_backends=[dict(type='LocalVisBackend')],
    name='visualizer')
log_processor = dict(type='LogProcessor', window_size=50, by_epoch=True)
log_level = 'INFO'
load_from = None
resume = False
custom_hooks = [
    dict(type='NumClassCheckHook'),
    dict(type='CheckInvalidLossHook', interval=50, priority='VERY_LOW')
]
launcher = 'none'
work_dir = './work_dirs/ssd300_voc0712'

Result has been saved to /home/franchesoni/mine/creations/phd/projects/current/iadet/mmdetection/work_dirs/ssd300_voc0712/modules_statistic_results.json
10/31 14:17:03 - mmengine - INFO - Distributed training is not used, all SyncBatchNorm (SyncBN) layers in the model will be automatically reverted to BatchNormXd layers if they are used.
Traceback (most recent call last):
  File "/home/franchesoni/mine/creations/phd/projects/current/iadet/env_iadet/lib/python3.10/site-packages/mmengine/registry/build_functions.py", line 121, in build_from_cfg
    obj = obj_cls(**args)  # type: ignore
  File "/home/franchesoni/mine/creations/phd/projects/current/iadet/env_iadet/lib/python3.10/site-packages/mmengine/dataset/dataset_wrapper.py", line 52, in __init__
    raise ValueError(
ValueError: The meta information of the 2-th dataset does not match meta information of the first dataset

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/franchesoni/mine/creations/phd/projects/current/iadet/env_iadet/lib/python3.10/site-packages/mmengine/registry/build_functions.py", line 121, in build_from_cfg
    obj = obj_cls(**args)  # type: ignore
  File "/home/franchesoni/mine/creations/phd/projects/current/iadet/env_iadet/lib/python3.10/site-packages/mmengine/dataset/dataset_wrapper.py", line 183, in __init__
    self.dataset = DATASETS.build(dataset)
  File "/home/franchesoni/mine/creations/phd/projects/current/iadet/env_iadet/lib/python3.10/site-packages/mmengine/registry/registry.py", line 421, in build
    return self.build_func(cfg, *args, **kwargs, registry=self)
  File "/home/franchesoni/mine/creations/phd/projects/current/iadet/env_iadet/lib/python3.10/site-packages/mmengine/registry/build_functions.py", line 135, in build_from_cfg
    raise type(e)(
ValueError: class `ConcatDataset` in mmengine/dataset/dataset_wrapper.py: The meta information of the 2-th dataset does not match meta information of the first dataset

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/franchesoni/mine/creations/phd/projects/current/iadet/env_iadet/lib/python3.10/site-packages/mmengine/registry/build_functions.py", line 121, in build_from_cfg
    obj = obj_cls(**args)  # type: ignore
  File "/home/franchesoni/mine/creations/phd/projects/current/iadet/env_iadet/lib/python3.10/site-packages/mmengine/runner/loops.py", line 43, in __init__
    super().__init__(runner, dataloader)
  File "/home/franchesoni/mine/creations/phd/projects/current/iadet/env_iadet/lib/python3.10/site-packages/mmengine/runner/base_loop.py", line 26, in __init__
    self.dataloader = runner.build_dataloader(
  File "/home/franchesoni/mine/creations/phd/projects/current/iadet/env_iadet/lib/python3.10/site-packages/mmengine/runner/runner.py", line 1337, in build_dataloader
    dataset = DATASETS.build(dataset_cfg)
  File "/home/franchesoni/mine/creations/phd/projects/current/iadet/env_iadet/lib/python3.10/site-packages/mmengine/registry/registry.py", line 421, in build
    return self.build_func(cfg, *args, **kwargs, registry=self)
  File "/home/franchesoni/mine/creations/phd/projects/current/iadet/env_iadet/lib/python3.10/site-packages/mmengine/registry/build_functions.py", line 135, in build_from_cfg
    raise type(e)(
ValueError: class `RepeatDataset` in mmengine/dataset/dataset_wrapper.py: class `ConcatDataset` in mmengine/dataset/dataset_wrapper.py: The meta information of the 2-th dataset does not match meta information of the first dataset

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/franchesoni/mine/creations/phd/projects/current/iadet/mmdetection/tools/train.py", line 120, in <module>
    main()
  File "/home/franchesoni/mine/creations/phd/projects/current/iadet/mmdetection/tools/train.py", line 116, in main
    runner.train()
  File "/home/franchesoni/mine/creations/phd/projects/current/iadet/env_iadet/lib/python3.10/site-packages/mmengine/runner/runner.py", line 1629, in train
    self._train_loop = self.build_train_loop(
  File "/home/franchesoni/mine/creations/phd/projects/current/iadet/env_iadet/lib/python3.10/site-packages/mmengine/runner/runner.py", line 1444, in build_train_loop
    loop = LOOPS.build(
  File "/home/franchesoni/mine/creations/phd/projects/current/iadet/env_iadet/lib/python3.10/site-packages/mmengine/registry/registry.py", line 421, in build
    return self.build_func(cfg, *args, **kwargs, registry=self)
  File "/home/franchesoni/mine/creations/phd/projects/current/iadet/env_iadet/lib/python3.10/site-packages/mmengine/registry/build_functions.py", line 135, in build_from_cfg
    raise type(e)(
ValueError: class `EpochBasedTrainLoop` in mmengine/runner/loops.py: class `RepeatDataset` in mmengine/dataset/dataset_wrapper.py: class `ConcatDataset` in mmengine/dataset/dataset_wrapper.py: The meta information of the 2-th dataset does not match meta information of the first dataset
```

</details>

### Additional information

After some investigation, I think the problem lies in the following lines of `mmdet/datasets/voc.py`

https://github.com/open-mmlab/mmdetection/blob/3221ecf33885bcf185e1ea63ac72a861403d81b0/mmdet/datasets/voc.py#L26-L31 

which causes the metadata to be different and metainfo consistency check to fail (linked below)

https://github.com/open-mmlab/mmengine/blob/64ac14303fc786151beabba1de4ddfd7c2e5a408/mmengine/dataset/dataset_wrapper.py#L48-L54

**Comment by Czm369** [comment]
This is a bug for `ConcatDataset` in` mmengine/dataset/dataset_wrapper.py`，you can delete 26-31 to avoid it.
https://github.com/open-mmlab/mmdetection/blob/3221ecf33885bcf185e1ea63ac72a861403d81b0/mmdet/datasets/voc.py#L26

**Comment by minipuding** [comment]
This problem is still not fixed at 3.0.0rc6
